go/libraries/doltcore/table/editor/pk_table_editor.go

```diff
@@ -152,6 +166,45 @@ func newTableEditAcc(nbf *types.NomsBinFormat) *tableEditAccumulator {
 	}
 }
 
+// CreateFromCurrent returns a new tableEditAccumulator that references the current tableEditAccumulator.
+func (tea *tableEditAccumulator) CreateFromCurrent() *tableEditAccumulator {
+	return &tableEditAccumulator{
+		prevTea:      tea,
+		prevTeaAsMap: types.EmptyMap,
+		nbf:          tea.nbf,
+		ed:           types.CreateEditAccForMapEdits(tea.nbf),
+		insertedKeys: make(map[hash.Hash]types.Value),
+		addedKeys:    make(map[hash.Hash]types.Value),
+		removedKeys:  make(map[hash.Hash]types.Value),
+		affectedKeys: make(map[hash.Hash]types.Value),
+	}
+}
+
+// Has returns whether the current tableEditAccumulator contains the given key. This assumes that the given hash is for the given
+// key.
+func (tea *tableEditAccumulator) Has(ctx context.Context, keyHash hash.Hash, key types.Value) (bool, error) {
+	if _, ok := tea.addedKeys[keyHash]; ok {
+		return true, nil
+	}
+	if _, ok := tea.removedKeys[keyHash]; !ok {
+		// When prevTeaAsMap is updated, prevTea is set to nil. Therefore, if prevTea is non-nil, we use it.
+		if tea.prevTea != nil {
```

Comment:
You need a lock here to avoid a race with prevTea and prevTeaAsMap being updated.

Reply:
There are locks around all of the calls and modifications, they're just done from outside of the struct methods. I'll add a note.